src/apps/Permissions/PermissionsView.js

```diff
@@ -0,0 +1,170 @@
+import React, { useCallback, useMemo } from 'react'
+import PropTypes from 'prop-types'
+import {
+  ButtonIcon,
+  ContextMenu,
+  ContextMenuItem,
+  DataView,
+  IconTrash,
+  textStyle,
+  useLayout,
+  useTheme,
+} from '@aragon/ui'
+import { usePermissions } from '../../contexts/PermissionsContext'
+import LocalIdentityBadge from '../../components/IdentityBadge/LocalIdentityBadge'
+import PermissionsIdentityBadge from './PermissionsIdentityBadge'
+
+function PermissionsView({ permissions, onManageRole, heading, showApps }) {
+  const { layoutName } = useLayout()
+  const willRenderEntryChild = useMemo(
+    () => permissions.some(permission => permission.entities.length > 1),
+    [permissions]
+  )
+
+  const fields = [
+    'Action',
+    'On app',
+    { label: 'Assigned to entity', childStart: true },
+    'Managed by',
+  ]
+
+  if (!showApps) {
+    fields.splice(1, 1)
+  }
+
+  return (
+    <DataView
+      heading={heading}
+      mode={
+        layoutName === 'large' || (layoutName === 'medium' && !showApps)
+          ? 'table'
+          : 'list'
+      }
+      fields={permissions.length ? fields : []}
+      entries={permissions}
+      renderEntry={entry => renderEntry(entry, showApps)}
+      renderEntryChild={willRenderEntryChild ? renderEntryChild : undefined}
+      renderEntryActions={entry => (
+        <EntryActions entry={entry} onManageRole={onManageRole} />
+      )}
+    />
+  )
+}
+
+PermissionsView.propTypes = {
+  heading: PropTypes.node,
+  onManageRole: PropTypes.func.isRequired,
+  permissions: PropTypes.array.isRequired,
+  showApps: PropTypes.bool.isRequired,
+}
+
+function renderEntry({ entities, app, role, manager }, showApps) {
+  const cells = [
+    <span
+      css={`
+        ${textStyle('body2')}
+      `}
+    >
+      {role.name}
+    </span>,
+    <LocalIdentityBadge entity={app.proxyAddress} shorten />,
+    <EntryEntities entities={entities} />,
+    <PermissionsIdentityBadge entity={manager} />,
+  ]
+
+  if (!showApps) {
+    cells.splice(1, 1)
+  }
+
+  return cells
+}
+
+function renderEntryChild({ entities, app, role }) {
+  return entities.length < 2
+    ? null
+    : entities.map(entity => (
+        <ChildEntity
+          key={entity.address}
+          appAddress={app.proxyAddress}
+          entity={entity}
+          roleBytes={role.bytes}
+        />
+      ))
+}
+
+/* eslint-disable react/prop-types */
```

Comment:
For the time being or forever?

Reply:
Forever IMO, I’d like to disable this check for internal components but it’s not possible with this ESLint rule :-(

Reply:
Why not set the `PropTypes` for the inner components? i do it and imo seems cleaner that disabling `eslint` checks

Reply:
I don’t find it that useful when the component using it is right here, in the same file. I like to have inner components easy to manipulate / refactor, like other inner functions which don’t require such a thing (at least until we use something like TypeScript or Flow :laughing:).

But I’m fine with it if we decide to add prop types everywhere, it’s really a personal opinion.